File: Javascript/Ajax/vb.net/App_Code/rss.vb
Imports System.Collections.Generic
Imports System.ServiceModel.Syndication
Imports System.Xml

Public Structure RSSItem
    Private _title As String
    Public Property title() As String
        Get
            Return _title
        End Get
        Set(ByVal value As String)
            _title = value
        End Set
    End Property

    Private _link As String
    Public Property link() As String
        Get
            Return _link
        End Get
        Set(ByVal value As String)
            _link = value
        End Set
    End Property

    Private _description As String
    Public Property description() As String
        Get
            Return _description
        End Get
        Set(ByVal value As String)
            _description = value
        End Set
    End Property
End Structure

Public Class RSSReader

    Public title As String
    Public link As String
    Public description As String
    Public items As List(Of RSSItem) = New List(Of RSSItem)()

    Public Sub New(ByVal url As String)

        Dim doc As New XmlDocument()
        doc.Load(url)

        Dim channel As XmlElement = doc("rss")("channel")
        Dim items As XmlNodeList = channel.GetElementsByTagName("item")
        Me.title = channel("title").InnerText
        Me.link = channel("link").InnerText
        Me.description = channel("description").InnerText

        For Each item As XmlNode In items
            Dim rss As New RSSItem()
            rss.title = item("title").InnerText
            rss.description = item("description").InnerText
            rss.link = item("link").InnerText
            Me.items.Add(rss)
        Next

    End Sub
End Class

File: Javascript/Ajax/csharp/App_Code/rss.cs
using System.Collections.Generic;
using System.Xml;

public struct RSSItem
{
    private string _title;
    public string title
    {
        get
        {
            return _title;
        }
        set
        {
            _title = value;
        }
    }

    private string _link;
    public string link
    {
        get
        {
            return _link;
        }
        set
        {
            _link = value;
        }
    }

    private string _description;
    public string description
    {
        get
        {
            return _description;
        }
        set
        {
            _description = value;
        }
    }
}


public class RSSReader
{
    public string title;
    public string link;
    public string description;
    public List<RSSItem> items = new List<RSSItem>();

    public RSSReader(string url)
    {
        XmlDocument doc = new XmlDocument();
        doc.Load(url);

        XmlElement channel = doc["rss"]["channel"];
        XmlNodeList items = channel.GetElementsByTagName("item");
        this.title = channel["title"].InnerText;
        this.link = channel["link"].InnerText;
        this.description = channel["description"].InnerText;

        foreach (XmlNode item in items)
        {
            RSSItem rssItem = new RSSItem();
            rssItem.title = item["title"].InnerText;
            rssItem.description = item["description"].InnerText;
            rssItem.link = item["link"].InnerText;
            this.items.Add(rssItem);
        }
    }
}
